pkg/queue/manager.go

```diff
@@ -547,3 +552,27 @@ func (m *Manager) reportPendingWorkloads(cqName string, cq ClusterQueue) {
 	}
 	metrics.ReportPendingWorkloads(cqName, active, inadmissible)
 }
+
+func (m *Manager) GetClusterQueues() map[string]ClusterQueue {
+	m.RLock()
+	defer m.RUnlock()
+	return m.clusterQueues
+}
+
+func (m *Manager) SetSnapshot(cqName string, workloads []kueue.ClusterQueuePendingWorkload) {
```

Comment:
This is probably not needed if we have `UpdateSnapshot` which computes and saves the snapshot.